marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.tsx

```diff
@@ -620,12 +620,12 @@ const SourceToEntityMap = (props) => {
                     autoSize={{ minRows: 1 }}
                     disabled={!props.canReadWrite}></TextArea>&nbsp;&nbsp;
                 <span>
-                    <Dropdown overlay={sourceSearchMenu} trigger={['click']} >
+                    <Dropdown overlay={sourceSearchMenu} trigger={['click']} disabled={!props.canReadWrite}>
                         <i  id="listIcon"><FontAwesomeIcon icon={faList} size="lg"  data-testid={row.name.split('/').pop()+'-listIcon'}  className={styles.listIcon} onClick={(e) => handleSourceList(row)}/></i>
                     </Dropdown>
                 </span>
                 &nbsp;&nbsp;
-                <span ><Dropdown overlay={menu} trigger={['click']}><Button id="functionIcon" data-testid={`${row.name.split('/').pop()}-${row.key}-functionIcon`} className={styles.functionIcon} size="small" onClick={(e) => handleFunctionsList(row.name)}>fx</Button></Dropdown></span></div>
+                <span ><Dropdown overlay={menu} trigger={['click']} disabled={!props.canReadWrite}><Button id="functionIcon" data-testid={`${row.name.split('/').pop()}-${row.key}-functionIcon`} className={styles.functionIcon} size="small" onClick={(e) => handleFunctionsList(row.name)}>fx</Button></Dropdown></span></div>
```

Comment:
Is there not a way to add a test for this change?

Reply:
I am not sure but I will try to write test for it and submit PR again

Reply:
In the test 'Verify evaluation of valid expression for mapping reader user' in source-to-entity-map.test you can add couple of assertions for functionIcon and listIcon that they are disabled.